sopel/tools/target.py

```diff
@@ -107,9 +108,193 @@ def add_user(self, user, privs=0):
+        return self.privileges.get(Identifier(nick), 0) & plugin.OPER
+
+    def is_owner(self, nick):
+        """Tell if a user has the OWNER privilege level.
+
+        :param str nick: a user's nick in this channel
+        :rtype: bool
+
+        Unlike :meth:`has_privilege`, this method checks if the user has been
+        explicitly granted the OWNER privilege level::
+
+            >>> channel.add_user(some_user, plugin.OWNER)
+            >>> channel.is_owner(some_user.nick)
+            True
+            >>> channel.is_voiced(some_user.nick)
+            False
+
+        Note that you can always have more than one privilege level::
+
+            >>> channel.add_user(some_user, plugin.OWNER | plugin.VOICE)
+            >>> channel.is_owner(some_user.nick)
+            True
+            >>> channel.is_voiced(some_user.nick)
+            True
+
+        """
+        return self.privileges.get(Identifier(nick), 0) & plugin.OWNER
+
+    def is_admin(self, nick):
+        """Tell if a user has the ADMIN privilege level.
+
+        :param str nick: a user's nick in this channel
+        :rtype: bool
+
+        Unlike :meth:`has_privilege`, this method checks if the user has been
+        explicitly granted the ADMIN privilege level::
+
+            >>> channel.add_user(some_user, plugin.ADMIN)
+            >>> channel.is_admin(some_user.nick)
+            True
+            >>> channel.is_voiced(some_user.nick)
+            False
+
+        Note that you can always have more than one privilege level::
+
+            >>> channel.add_user(some_user, plugin.ADMIN | plugin.VOICE)
+            >>> channel.is_admin(some_user.nick)
+            True
+            >>> channel.is_voiced(some_user.nick)
+            True
+
+        """
+        return self.privileges.get(Identifier(nick), 0) & plugin.ADMIN
+
+    def is_op(self, nick):
+        """Tell if a user has the OP privilege level.
+
+        :param str nick: a user's nick in this channel
+        :rtype: bool
+
+        Unlike :meth:`has_privilege`, this method checks if the user has been
+        explicitly granted the OP privilege level::
+
+            >>> channel.add_user(some_user, plugin.OP)
+            >>> channel.is_op(some_user.nick)
+            True
+            >>> channel.is_voiced(some_user.nick)
+            False
+
+        Note that you can always have more than one privilege level::
+
+            >>> channel.add_user(some_user, plugin.OP | plugin.VOICE)
+            >>> channel.is_op(some_user.nick)
+            True
+            >>> channel.is_voiced(some_user.nick)
+            True
+
+        """
+        return self.privileges.get(Identifier(nick), 0) & plugin.OP
+
+    def is_halfop(self, nick):
+        """Tell if a user has the HALFOP privilege level.
+
+        :param str nick: a user's nick in this channel
+        :rtype: bool
+
+        Unlike :meth:`has_privilege`, this method checks if the user has been
+        explicitely granted the HALFOP privilege level::
```

Comment:
I missed another typo 😩

```suggestion
        explicitly granted the HALFOP privilege level::
```